Assets/Scripts/Communication/CommunicationManager.cs

```diff
@@ -0,0 +1,392 @@
+using UnityEngine;
+using System;
+using System.Threading;
+using System.Collections.Generic;
+using Common.Cryptography;
+using System.Security.Cryptography;
+using System.Text.RegularExpressions;
+using System.Linq;
+using System.Text;
+
+public class CommunicationManager
+{
+	private static Thread _manager;
+	private static bool _threadRunning;
+
+	private static object asyncMessagesLock;
+	private static object requestQueueLock;
+	private static object responseDictLock;
+
+	private static Queue<Dictionary<string, object>> asyncMessagesQueue;
+	private static Queue<Dictionary<string, object>> requestQueue;
+	private static Dictionary<string, Dictionary<string, object>> responseDict;
+
+	private static string url = "ws://tactics-dev.ddns.net:8443";
+	//private static string url = "ws://localhost:8000";
+	//private static string url = ""ws://tactics-production.herokuapp.com/""
+
+	public static void Start()
+	{
+		Debug.Log("Starting Communication Manager");
+		//Create the communication data structures
+		asyncMessagesQueue = new Queue<Dictionary<string, object>>();
+		requestQueue = new Queue<Dictionary<string, object>>();
+		responseDict = new Dictionary<string, Dictionary<string, object>>();
+
+		//Create the communication lock objects
+		asyncMessagesLock = new object();
+		requestQueueLock = new object();
+		responseDictLock = new object();
+
+		//Create the server connection
+		Connect();
+
+		//Start the communicator thread
+		_manager = new Thread(ProcessRequests);
+		_manager.Start();
+		Debug.Log("Communication Manager has started successfully");
+	}
+
+	static void ProcessRequests(){
+		_threadRunning = true;
+
+		try{
+			while (_threadRunning){
```

Comment:
Unfortunately, I did not put much time looking into the notification type mechanism.  With our current setup, I don't know how to have it sleep until a response comes in.  We could add an arbitrary sleep amount but then we trade off speed of the app getting data from the server.  I haven't tried it on a phone so I can't say how this will perform.